docs/src/parallel_hpc.md

```diff
@@ -0,0 +1,106 @@
+
+Call julia with prespecified number of threads, e.g, 
+```
+$ julia --project -t 16
+```
+Then in the example use the following loop:
+```julia
+Threads.@threads for i = 1:N_ens
+    ...
+    g_ens[:, i] = lorenz_forward(settings, lorenz_params[i])
+end
+```
+Read more about multi-threading [here](https://docs.julialang.org/en/v1/manual/multi-threading/).
+
+### Parallel map `pmap`
+When using multiple processes - the package managment must also be loaded on the worked processors. To call julia `--project` with 4 worker processes, we include the line inside
+```julia
+using Distributed
+addprocs(4;exeflags="--project")
+```
+We use  `pmap` to apply a function to each element of a list (in this case a list of different parameter configurations of the forward model). 
+```julia
+g_ens = pmap (x -> lorenz_forward(settings, x), lorenz_params]
+```
+If `pmap` is used within another module, that module will also need to be loaded on all workers. For this we use the macro `@everywhere module XYZ`.
+
+Read more about `pmap` [here](https://docs.julialang.org/en/v1/manual/distributed-computing/#Parallel-Map-and-Loops).
+
+### Distributed loop `@distributed for` 
+When using multiple processes - the package managment must also be loaded on the worked processors. To call julia `--project` with 4 worker processes, we include the line inside
+```julia
+using Distributed
+addprocs(4;exeflags="--project")
+```
+In the code we then can use [SharedArrays](https://docs.julialang.org/en/v1/manual/distributed-computing/#man-shared-arrays) with a distributed loop.
+```julia
+using SharedArrays
+
+g_ens = SharedArray{Float64}(dim_data,N_ens)
+@sync @distributed for i=1:N_ens
+    ...
+    g_ens[:,i] = lorenz_forward(settings,lorenz_params[i])
+end
+```
+`@sync` forces the code to wait until all processes in the `@distributed for` loop are complete before continuing.
+
+If `@distributed` is used within another module, that module will also need to be loaded on all workers. For this we use the macro `@everywhere module XYZ`.
+
+!!! note
+    `@distributed for` is most performant when one has large numbers of iterations (i.e. large `N_ens`) and the inner loop is computationally tiny (i.e. `lorenz_forward` is cheap). In the ensemble setting, `pmap` is usually the preferred choice.
+Read more about `@distributed` [here](https://docs.julialang.org/en/v1/manual/distributed-computing/#Parallel-Map-and-Loops)
+### Case 2: HPC interface
+For more intensive applications, such as found when interfacing with non-Julia codes and HPC workload managers, we suggest using an alternative workflow where one interleaves scripts that launch EKP updates, and model runs. One possible implementation is the following loop
+- Step `0`(a). Write an ensemble of parameter files `parameters_0_i` for `i=1:N_ens`, with each parameter file containing a sample from the prior distribution.
+- Step `0`(b). Setup EKP EKP object and save in `jld2` format e.g. `ekpobject.jld2`.
+
+for `n=0,...,N_it`, do:
```

Comment:
we want the `n` index to line up. I called the initial ensemble `*_0_i` so this is used. I've adjusted to using `n=1:N_ens` at the cost of needing `{n-1}` indices instead of `n`